test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_smdp.py

```diff
@@ -72,45 +75,18 @@ def get_transformers_version(ecr_image):
     else:
         raise LookupError("HF transformers version not found in image URI")
 
+
 @pytest.mark.integration("smdataparallel")
 @pytest.mark.model("hf_qa_smdp")
 @pytest.mark.processor("gpu")
 @pytest.mark.skip_cpu
 @pytest.mark.skip_py2_containers
-def test_smdp_question_answering(ecr_image, instance_type, py_version, sagemaker_session, tmpdir):
+def test_smdp_question_answering(ecr_image, sagemaker_regions, py_version, ):
     """
     Tests SM Distributed DataParallel single-node via script mode
     """
-    transformers_version = get_transformers_version(ecr_image)
-    git_config = {'repo': 'https://github.com/huggingface/transformers.git', 'branch': 'v'+transformers_version}
-
-    validate_or_skip_smdataparallel(ecr_image)
-    instance_count = 1
-    instance_type = "ml.p3.16xlarge"
-    
-    source_dir = (
-        "./examples/question-answering"
-        if Version(transformers_version) < Version("4.6")
-        else "./examples/pytorch/question-answering"
-    )
-    
-    
-    with timeout(minutes=DEFAULT_TIMEOUT):
-        estimator = HuggingFace(
-            entry_point='run_qa.py',
-            source_dir=source_dir,
-            git_config=git_config,
-            metric_definitions=metric_definitions,
-            role='SageMakerRole',
-            image_uri=ecr_image,
-            instance_count=instance_count,
-            instance_type=instance_type,
-            sagemaker_session=sagemaker_session,
-            py_version=py_version,
-            distribution=distribution,
-            hyperparameters=hyperparameters,
-        )
-        estimator.fit(job_name=sagemaker.utils.unique_name_from_base('test-hf-pt-qa-smdp'))
+    invoke_hf_pt_helper_function(ecr_image, sagemaker_regions, _test_smdp_question_answering_function,
+                                 py_version, 1)
```

Comment:
function args should be used for better maintainability and readability
```c
function_args = {
            'py_version': py_version,
            'instances_quantity': 1,
        }
        
invoke_hf_pt_helper_function(ecr_image, sagemaker_regions, _test_smdp_question_answering_function, function_args)
```
and

Reply:
As we agreed it will require quite a lot of work but won't add too much value since helper functions are literally 2 raws below.